### ISSUE TITLE
[Bug]: Use of aligned_textgrid.interleave_class() changes changes class definition causing bugs when iterating

### ISSUE THREAD

**Issue by chrisbrickhouse** [Forced-Alignment-and-Vowel-Extraction/fave-syllabify#5]
### What happened?

I'll edit this with a minimal example later today, but the gist is that if you use `fave_syllabify.syllabify_tg()` while looping over multiple text grids, the first one works fine but then all others fail.

I tracked the culprit down to the use of `aligned_textgrid.interleave_class()` on [line 183](https://github.com/Forced-Alignment-and-Vowel-Extraction/fave-syllabify/blob/dev/src/fave_syllabify/syllabify.py#L183) and [line 195](https://github.com/Forced-Alignment-and-Vowel-Extraction/fave-syllabify/blob/dev/src/fave_syllabify/syllabify.py#L195). This causes the `Phone` and `Word` classes to change their superset-subset relationship which messes up future iterations that assume these classes are superset-subsets. Ideally these classes and functions shouldn't be so tightly coupled.

A workaround is to call a cleanup function at the end of every iteration that resets the relationship.

### What operating system are you using?

Ubuntu

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by JoFrhwld** [comment]
Oh gosh you're right, that would cause problems!

Maybe another approach would be to initialize a vacuous subclass for entry classes for each AlignedTextGrid object? So, whatever gets passed to the entry classes argument, the actual entry class is:

```
class Word_local(Word):
  def __init__(self, *args, **kwargs):
    super().__init__(*args, **kwargs)
```

This will keep the entry classes isolated within each textgrid, but should keep any `isinstance()` checks working?

How exactly it could work might involve copying some of the class factory code from `custom_classes()`

**Comment by JoFrhwld** [comment]
Started making some progress [here](https://github.com/Forced-Alignment-and-Vowel-Extraction/alignedTextGrid/compare/dev...180-bug-insulate-entry-classes-within-alignedtextgrid-instances-from-the-global-class)

**Comment by JoFrhwld** [comment]
Just published v0.1.3 to pypi. Should have this issue resolved
